src/Turtle/Prelude.hs

```diff
@@ -1095,6 +1099,33 @@ time io = do
 hostname :: MonadIO io => io Text
 hostname = liftIO (fmap Text.pack getHostName)
 
+-- | Show the full path of an executable file
+which :: MonadIO io => FilePath -> io (Maybe FilePath)
+which cmd = fold (whichAll cmd) Control.Foldl.head
+
+-- | Show all matching executables in PATH, not just the first
+whichAll :: FilePath -> Shell FilePath
+whichAll cmd =
+    need "PATH" >>= \case
```

Comment:
`Shell` handles pattern failures as `empty`, so this line can be written as `Just paths <- need "PATH"`, removing another indentation level.